app/views/runs/_timeline.slim

```diff
@@ -4,12 +4,13 @@
 - else
   - if run.video&.supports_embedding?
     div style="height: 1em": span.video-progress-line.timeline-animation id="video-progress-line-#{run.id36}"
-  .timeline-background.timeline-animation data={run_id: run.id36}
+  - collapsed_segments = run.collapsed_segments(timing)
```

Comment:
Is there a benefit to caching this manually in a variable here vs. letting Rails cache the query for us?

Reply:
`ForgetfulPersonsRun.collapsed_segments` does do a non-zero amount of work apart from just a SQL query. Since I was adding additional calls to it, it seemed sensible to go ahead and just cache the result. It does look like some of those calls will be going away, though.

Reply:
Oh that's right, I forgot that method is not just a query!

Reply:
2 of the uses of `collapsed_segments` were removed leaving just 2. I'm fine going back to 2 calls of the original if you'd prefer. The performance difference is probably negligible.